drivers/spi/opentitan/driver.h

```diff
@@ -0,0 +1,118 @@
+#include <stdint.h>
+#include <stdbool.h>
+
+typedef struct spi_driver_data {
+    // Per request state
+    spi_cs_line_t cs_line;
+    spi_cmd_t *cmd;
+    uint16_t num_cmds;
+    void *slice_base;
+    // Request in-progress state
+    uint8_t cmd_in_progress;
+    // Command in-progress state (effectively offsets into the slice region)
+    uint16_t tx_remaining;
+    uint16_t rx_remaining;
+    // Error
+    spi_err_t err;
+} spi_driver_data_t;
+
+typedef enum spi_state {
+    IDLE, REQ, SEL_CMD, CMD, CMD_RET, RESP, NUM_STATES
+} spi_state_t;
+
+typedef struct fsm_state {
+    spi_state_t nxt_state;
+    bool yield;
+} fsm_state_t;
+
+static void spi_reset_state(spi_driver_data_t *s) {
+    s->cmd = NULL;
```

Comment:
nitpick: doesn't hurt to make sure the order of members in the reset function matches the struct definition. makes it easier for people maintaining this later.